programs/survival/potion_master_cleric.sc

```diff
@@ -130,15 +130,60 @@ global_potion_trades = [
 	
 ];
 
+// potentially override chosen max trades to prevent duplicates
+global_max_potion_trades = min(global_max_potion_trades,length(global_potion_trades) * length(global_potion_types));
+
 _add_trades(cleric, p, nbt) -> (
+	trades = {};
+	effects = copy(global_potion_trades);
 	// add 0-max number of potion trades
 	loop(rand(1 + global_max_potion_trades),
-
+		i = _;
+		mapped_effects = map(effects, _ ~ '(?<=Potion of )\\w*');
```

Comment:
> Why did you take the time to map the effects instead of using them directly?

Because debugging with a map that has keys that are 12+ lines of text is painful.